api/src/main/scala/hmda/api/http/ValidationErrorConverter.scala

```diff
@@ -62,13 +70,29 @@ trait ValidationErrorConverter {
       .getOrElse("")
   }
 
-  private def relevantFields(err: ValidationError): JsObject = {
+  private def relevantRow(err: ValidationError, vs: HmdaFileValidationState): HmdaFileRow = {
+    if (err.ts) vs.ts.getOrElse(HmdaRowError())
+    else vs.lars.find(lar => lar.loan.id == err.errorId).getOrElse(HmdaRowError())
+  }
+
+  private def relevantFields(err: ValidationError, row: HmdaFileRow): JsObject = {
     val fieldNames: Seq[String] = editDescriptions.find(e => e.editNumber == err.ruleName)
```

Comment:
You're right. This is much more streamlined. Updated in 2d0a294395e50d3d37b25536e172fbf6e9058975.